test/parallel/test-tls-basic-validations.js

```diff
@@ -102,3 +102,16 @@ common.expectsError(
     assert(out.ALPNProtocols.equals(Buffer.from(expectView)));
   }
 }
+
+{
+  const protocols = [(new String('a')).repeat(500)];
+  const out = {};
+  common.expectsError(
+    () => tls.convertALPNProtocols(protocols, out),
+    {
+      code: 'ERR_OUT_OF_RANGE',
+      message: 'The value of "byte length of a protocol" is out ' +
+        'of range. It must be <= 255. Received 500'
```

Comment:
The message here seems somewhat confusing. I have no good alternative right now but I would like to iterate over this a bit further. What about `The value of "one of the protocols byte length" is out of range...`? It is not ideal either because of the quotation marks but for me it's better readable. But we might just use a different error class instead?

Reply:
To be honest, I'm in agreement. I think it reads clunkily right now, and it's still clunky with your suggested wording. Another error class might be appropriate, but I went with `ERR_OUT_OF_RANGE` because it was the suggested one. Which other error class would you choose?

Reply:
Seems like we should do one of two things:

* Update the ERR_OUT_OF_RANGE message to make it more generic. Or allow an override of the entire message for situations like this?

* Create a new error code, like ERR_LENGTH_OUT_OF_RANGE.

Reply:
(By the way, I'm fine with this landing and we iterate on the message in other PRs. But I'm of course also fine with it being improved in this PR!)

Reply:
I like the idea of allowing an override of the message. If I changed the definition for ERR_OUT_OF_RANGE to the following, would that be acceptable?

```javascript
E('ERR_OUT_OF_RANGE',
  (name, range, value, msg = null) => {
    msg = msg ? msg : `The value of "${name}" is out of range.`;
    if (range !== undefined) msg += ` It must be ${range}.`;
    msg += ` Received ${value}`;
    return msg;
  }, RangeError);
```

Reply:
I think that (or creating a new code) should be fine. The function signature can probably be improved a bit, but that may involve changing a lot of code that calls this function. Someone (you or someone else) can do that in a subsequent pull request rather than massively increase the diff on this one. Or it can simply not happen at all. TL;DR: SGTM!